tests/core/contracts/test_contract_init.py

```diff
@@ -11,12 +15,21 @@
 
 
 @pytest.fixture
-def math_addr(MathContract):
+def math_deploy_receipt(MathContract):
     web3 = MathContract.web3
     deploy_txn = MathContract.constructor().transact({'from': web3.eth.coinbase})
     deploy_receipt = web3.eth.waitForTransactionReceipt(deploy_txn)
     assert deploy_receipt is not None
-    return deploy_receipt['contractAddress']
+    return deploy_receipt
+
+
+@pytest.fixture(params=['hex', 'bytes'])
+def math_addr(request, math_deploy_receipt):
+    addr = math_deploy_receipt['contractAddress']
+    if request.param == 'bytes':
+        return to_bytes(hexstr=addr)
```

Comment:
I considered recommending `eth_utils.to_canonical_address(addr)` here, but I find it hard to remember that canonical means the `bytes` version (to me the semantically "canonical" version of the address is the checksum version).

So I actually like `to_bytes` for clarity here.